chains/substrate/executor/executor.go

```diff
@@ -79,22 +83,30 @@ func NewExecutor(
 }
 
 // Execute starts a signing process and executes proposals when signature is generated
-func (e *Executor) Execute(msgs []*message.Message) error {
+func (e *Executor) Execute(ctx context.Context, msgs []*message.Message) error {
 	e.exitLock.RLock()
 	defer e.exitLock.RUnlock()
+	spancContext, span := otel.Tracer("relayer-sygma").Start(ctx, "relayer.sygma.substrate.Execute")
+	defer span.End()
+	logger := log.With().Str("dd.trace_id", span.SpanContext().TraceID().String()).Logger()
 
 	proposals := make([]*chains.Proposal, 0)
 	for _, m := range msgs {
+		logger.Info().Str("msg.id", m.ID()).Msgf("Message to execute %s", m.String())
+		span.AddEvent("Message to execute received", traceapi.WithAttributes(attribute.String("msg.id", m.ID()), attribute.String("msg.type", string(m.Type))))
```

Comment:
Adding here all message properties (maybe implement message.String()) would be usefull.

Reply:
That make sense, added msg.String instead of type, but left msg.id since this tag can be used to search with